moveit_simple/src/moveit_simple.cpp

```diff
@@ -611,6 +619,50 @@ void Robot::clearTrajectory(const::std::string traj_name)
 
 
 
+moveit_simple::JointTrajectoryType Robot::plan(const std::string traj_name, 
+                                               bool collision_check)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  if ( traj_info_map_.count(traj_name) )
+  {
+    JointTrajectoryType goal;
+    goal.trajectory.joint_names = joint_group_->getVariableNames();
+    try
+    {
+      if ( toJointTrajectory(traj_name, goal.trajectory.points, collision_check) )
+      {
+
+        // Modify the speed of execution for the trajectory based off of the speed_modifier_
+        for (std::size_t i = 0; i < goal.trajectory.points.size(); i++)
+        {
+          goal.trajectory.points[i].time_from_start *= (1.0/speed_modifier_);
+        }
+
+        ROS_INFO_STREAM("Successfully planned out trajectory: [" << traj_name << "]");
+        return goal;
+      }
+      else
+      {
+        ROS_ERROR_STREAM("Failed to convert [" << traj_name << "] to joint trajectory");
+        throw IKFailException("Conversion to joint trajectory failed for " + traj_name);
+      }
+    }
+    catch(CollisionDetected &cd)
+    {
+      ROS_ERROR_STREAM("Collision detected in [" << traj_name << "]");
+      throw cd;
+    }
+  }
+  else
+  {
+    ROS_ERROR_STREAM("Trajectoy [" << traj_name << "] not found");
+    throw std::invalid_argument("No trajectory found named " + traj_name);
+  }
+}
+
+
+
 void OnlineRobot::execute(const std::string traj_name, bool collision_check)
```

Comment:
Why doesn't this method call `plan` followed by `execute` with the planned trajectory?